src/traces/contour/attributes.js

```diff
@@ -44,13 +44,15 @@ module.exports = extendFlat({}, {
     },
     ncontours: {
         valType: 'integer',
-        dflt: 0,
+        dflt: 15,
+        min: 1,
         role: 'style',
         description: [
             'Sets the maximum number of contour levels. The actual number',
             'of contours will be chosen automatically to be less than or',
             'equal to the value of `ncontours`.',
-            'Has an effect only if `autocontour` is *true*.'
+            'Has an effect only if `autocontour` is *true* or if',
+            '`contours.size` is missing.'
```

Comment:
So if you specify `ncontours` and `contours: {start, end}` we will automatically choose a "nice" size, even though `autocontour` is `false`. Any reason *not* to allow this usage?

Note that the converse is not allowed: you can't specify `autocontour=true` and use your own `contours.size` value instead of `ncontours` - though I suppose it *would* be clear what this means, as it uses the same machinery as axis ticks, ie it would use a canonical tick of 0 so the contours would be all the multiples of `contours.size` between the data min and max values. That would actually be kind of nice, maybe I'll add it as well.